lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -802,7 +801,7 @@ SPIRVType *LLVMToSPIRVBase::transSPIRVOpaqueType(StringRef STName,
 
 SPIRVType *LLVMToSPIRVBase::transScavengedType(Value *V) {
   if (auto *F = dyn_cast<Function>(V)) {
-    FunctionType *FnTy = F->getType()->isOpaquePointerTy()
+    FunctionType *FnTy = F->getType()->isPointerTy()
                              ? Scavenger->getFunctionType(F)
                              : F->getFunctionType();
```

Comment:
It seems like this can be replaced as follows:

```suggestion
    FunctionType *FnTy = Scavenger->getFunctionType(F);
```

Reply:
Indeed.